plugins/input/pgraster/pgraster_wkb_reader.cpp

```diff
@@ -193,8 +193,9 @@ void read_data_band(mapnik::raster_ptr raster,
 
   float* data = (float*)image.getBytes();
   double val;
-  val = reader(); // nodata value, need to read anyway
-  if ( hasnodata ) raster->set_nodata(val);
+  double nodataval;
+  nodataval = reader(); // nodata value, need to read anyway
+  if ( hasnodata ) raster->set_nodata(nodataval);
   for (int y=0; y<height; ++y) {
     for (int x=0; x<width; ++x) {
       val = reader();
```

Comment:
Right, it's a "data" raster, and any nodata transparency will have to be added by the rastercolorizer in the rendering engine itself.